metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_ui_aspects_generator.py

```diff
@@ -0,0 +1,156 @@
+import json
+import time
+
+from datahub import nice_version_name
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.mcp_builder import make_data_platform_urn
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.metadata.schema_classes import (
+    DataHubIngestionSourceConfigClass,
+    DataHubIngestionSourceInfoClass,
+    ExecutionRequestInputClass,
+    ExecutionRequestResultClass,
+    ExecutionRequestSourceClass,
+    _Aspect,
+)
+from datahub.utilities.urns.urn import Urn
+
+
+class DatahubIngestionUIHistoryAspectsGenerator:
+
+    _EXECUTOR_ID: str = "Datahub Ingestion CLI"
+    _TASK_NAME: str = "CLI Ingestion"
+    _EXECUTION_REQUEST_SOURCE_TYPE: str = "INGESTION_SOURCE"
+
+    @staticmethod
+    def get_cur_time_in_ms() -> int:
+        return int(time.time() * 1000)
+
+    def __init__(self, graph: DataHubGraph, ctx: PipelineContext) -> None:
+        def generate_entity_name() -> str:
```

Comment:
is there a reason to make this an embedded method vs just a normal instance method?